File: t/simple.cpp
#include <iostream>
#include "antlr4-runtime.h"
#include "vhdlLexer.h"
#include "vhdlParser.h"
#include "vhdlBaseListener.h"

using namespace vhdl;

int main(int argc, const char* argv[]) {

  std::ifstream stream;
  stream.open("iu3.vhd");
  ANTLRInputStream input(stream);
  vhdlLexer lexer(&input);
  CommonTokenStream tokens(&lexer);
  vhdlParser parser(&tokens);

  parser.design_file();

  return 0;
}

File: t/simple.js
const fs = require('fs');
var antlr4 = require('../antlr4-runtime-js/index');
var vhdlLexer = require('../lib/vhdlLexer.js');
var vhdlParser = require('../lib/vhdlParser.js');
var vhdlVisitor = require('../lib/vhdlVisitor.js');
var input = "LIBRARY IEEE; USE IEEE.std_logic_1164.all; package PACK1 is end PACK1;"

fs.readFile('t/iu3.vhd', 'utf8', function (err,data) {
    if (err) {
	return console.log(err);
    }
    input = data;
    var chars = new antlr4.InputStream(input);
    var lexer = new vhdlLexer.vhdlLexer(chars);
    var tokens  = new antlr4.CommonTokenStream(lexer);
    var parser = new vhdlParser.vhdlParser(tokens);
    var visitor = new vhdlVisitor.vhdlVisitor();
    //parser.buildParseTrees = true;
    var tree = parser.design_file();
    
});

//console.log(tree);
//visitor.visitDesign_file();
